$OpenBSD: patch-gthumb_make-header_sh,v 1.1 2010/10/01 08:14:15 jasper Exp $

Overhaul the script, as it would put 'n' instead of '\n'
in the template file.

--- gthumb/make-header.sh.orig	Thu Jul 15 20:45:39 2010
+++ gthumb/make-header.sh	Thu Sep 30 23:12:27 2010
@@ -1,11 +1,34 @@
 #!/bin/sh
 
-template="$1"
-
-shift
-includes="\n"
+includes=
 for i in "$@"; do
-	includes="${includes}#include <gthumb/"$i">\n"
+	includes="${includes} #include <gthumb/$i>\n"
 done
 
-sed -e 's|@@|'"$includes"'|' $template
+echo "/* Do not edit.  Autogenerated from src/make-header.sh */
+
+/*
+ *  GThumb
+ *
+ *  Copyright (C) 2009 The Free Software Foundation, Inc.
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Street #330, Boston, MA 02111-1307, USA.
+ */
+ 
+#ifndef GTHUMB_H
+#define GTHUMB_H
+$includes
+#endif /* GTHUMB_H */
+" > gthumb.h
